app/model/Dependency.scala

```diff
@@ -4,12 +4,24 @@ package model
   * A single dependency retrieved from an artifact.
   */
 sealed abstract class Dependency(val name: String,
-                                 val version: String)
+                                 val version: String) {
+  def getType: String
+}
 
 case class JvmDependency(override val name: String,
                          override val version: String)
-  extends Dependency(name, version)
+  extends Dependency(name, version) {
+  override def getType: String = "java"
+}
 
 case class NodeDependency(override val name: String,
                           override val version: String)
-  extends Dependency(name, version)
+  extends Dependency(name, version) {
+  override def getType: String = "nodejs"
+}
+
+case class YarnDependency(override val name: String,
```

Comment:
i'm wondering if having a yarn dependency is relevant, since a `Dependency` mostly represents the run-time of the dependency and not the dependency manager that imports it. This modification will break stuff in the `All Dependencies` tab

Reply:
Yes I agree. It's modified.

Reply:
thank you